### ISSUE TITLE
Fan speed settings for "internal" bridges do not work as expected

### ISSUE THREAD

**Issue by CageFox** [supermerill/Slic3r#1620]
The new version 2.3.57_1 still have strange effects when changing Infill bridges fan speed. When Infill bridges fan speed higher then the default all seems ok:

![изображение](https://user-images.githubusercontent.com/1295382/136547707-12a59c08-8c83-4340-b635-4a7a206417e4.png)

Buf if entered value is lower the new value do not work:

![изображение](https://user-images.githubusercontent.com/1295382/136547834-79f6b4cb-d3d3-4ba2-8666-8baf31c29261.png)

**Comment by CageFox** [comment]
> Did you read the tooltip for that setting ? It does state that it wont slow it down if its currently running at a HIGHER speed.

Yeah, and this IS the bug. There is already closed issue this bug fixed - but it not fixed. 

![изображение](https://user-images.githubusercontent.com/1295382/136555370-ee512ec0-a1e6-43a4-923d-e9bd2fca071a.png)

**Comment by CageFox** [comment]
> 
> 
> Wasn't aware of that issue, I was going by this discussion #1601 which I had read previously and what the actual software says its supposed to do. As the software is behaving exactly as its described in the tooltip then I wouldn't classify it as a bug.
> 
> If some users want it to work differently then that's a feature change request, not a bug. Given the contents of the discussion I posted then it seems we have a difference in user expectations on how they want it to work. That's a whole different problem.

https://github.com/supermerill/SuperSlicer/issues/1414

Original feature (fan speed for infill bridge) was suggested by me (link above) and it was implemented, thanx to author. But from the beginning it have strange idea: not to lower the speed. @supermerill maked fix and closed original issue, but fix not working, so I created new issue.

There must be way to specify any speed for fan in infill bridge, even 0 and even if this speed will be lower then regular fan speed.

**Comment by supermerill** [comment]
> Yeah, and this IS the bug. There is already closed issue this bug fixed - but it not fixed.

I fix the case where, when 'infill bridges' < 'bridges' AND 'infill bridges' < 'defautl speed', then the fan speed use the  'bridges' setting instead of the 'default speed' when computing the 'infill bridges'

iirc, I fix the bug describe here https://github.com/supermerill/SuperSlicer/issues/1414#issuecomment-917171275
I don't change the behaviour to 'erase' default speed.

I really don't see the point to lower the fan speed for bridges vs default. It can only create drawbacks, imo. Please provides some arguments to change my mind.

**Comment by supermerill** [comment]
> But this is not the bridge fan speed we talking about. This is fan speed of first top layer. And for materials like ABS I want to keep it at ZERO. But right now to keep it at zero I forced to set bridge fan speed at zero or default fan speed at zero

? I don't understand.
The title of this issue is `Fan speed settings for "internal" bridges do not work as expected`
not `Fan speed settings for "top" do not work as expected`
If you found another bug, I'm grateful for the report. Right now I'm a bit confused. What's wrong?

**Comment by supermerill** [comment]
> why can't I set infill bridge fan **lower than default**?

As the default speed can increase with low layer time. This ensures the fan won't slow down for bridges.
As a bridge is a feature where the best cooling is needed...

Top fan speed and perimeter speed are "free" because it's for an aesthetic thing.
Bridge cooling is to ensure minimal drop, an entirely different purpose.

**Comment by CageFox** [comment]
> 
> 
> > why can't I set infill bridge fan **lower than default**?
> 
> As a bridge is a feature where the best cooling is needed.

But this is not a bridge at all. I never understand while the first layer of top treated as bridge. No slicers excepr prusa and SS think of this layer as bridge and there is no need for this. And this setting we discussing is the way to allow slice this layer like this is top layer, not bridge layer

And summary: please make this setting be able to set any speed for fun, even lower then regular. And of course this must be able to set zero speed (not working now also)
There will be no harm in this, these who need no lower speed will not use the setting at all

**Comment by supermerill** [comment]
It's a bridge because there is no solid area under it, so nothing to flatten the filament.

After the merge with prusaslicer 2.4, you will have more settings for bridge control.

**Comment by CageFox** [comment]
> 
> 
> It's a bridge because there is no solid area under it, so nothing to flatten the filament, so it's extruded round.

I disagree, only prusa slicer and SS thinks of it as of bridge. I personally think the bridge is only external part of model with no solid area below and slicers like Cura and Simplify think the same

But this is no problem if there is possibility to print this part the way I need it printed

Without this new option there was BIG problems using super slicer with materials like ABS on models with "real" bridges which need to be cooled. Now this is not so problematic, but ability to disable fan with this "fake bridge" is the must

**Comment by supermerill** [comment]
By setting it to -1 or below default, as you have done, there is no fan speed difference between 'internal bridge' and normal solid infill.

**Comment by CageFox** [comment]
> 
> 
> By setting it to -1 or below default, as you have done, there is no fan speed difference between 'internal bridge' and normal solid infill.

There is huge difference:

![изображение](https://user-images.githubusercontent.com/1295382/139499286-9cb2307f-68a9-406d-a57c-084f5ff1ef7e.png)

![изображение](https://user-images.githubusercontent.com/1295382/139499295-0f4df672-de2a-484a-9aa8-a99e333fcc15.png)

![изображение](https://user-images.githubusercontent.com/1295382/139499314-b40b5184-d469-49c4-9a30-273edb7fff7c.png)

**Comment by CageFox** [comment]
![изображение](https://user-images.githubusercontent.com/1295382/139499435-b1665b65-dec6-4c65-be06-3b189a5c0fed.png)

But

![изображение](https://user-images.githubusercontent.com/1295382/139499459-3e8695bf-35bb-40ef-8508-bcf87b334bd0.png)

Still 15% - fan not disabled as in description

**Comment by supermerill** [comment]
> Still 15%

that's what you want:

> I never understand while the first layer of top treated as bridge.

It's now treated as anything else (fan-wise), like if it wasn't a bridge.

**Comment by CageFox** [comment]
> 
> 
> > Still 15%
> 
> that's what (i understand that) you want:
> 
> > I never understand while the first layer of top treated as bridge.
> 
> It's now treated as anything else (fan-wise), like if it wasn't a bridge.

![изображение](https://user-images.githubusercontent.com/1295382/139500593-564ef9c6-6818-4a06-911d-8882e95cd1f1.png)

**Comment by supermerill** [comment]
![image](https://user-images.githubusercontent.com/6536403/139500658-05369f2a-24f0-426b-8a2b-497aa9b00380.png)

but ok, i'll rephrase it.

**Comment by CageFox** [comment]
> 
> 
> but ok, i'll rephrase it.

No, please, change it to work as in description al least. It is pain to print abs without this, I don't want to come back to Cura because of such small but deadly problem

**Comment by supermerill** [comment]
cura allow you to stop the fan for the first solid top layer?
what setting?

**Comment by CageFox** [comment]
> 
> 
> cura allow you to stop the fan for the first solid top layer? what setting?

Cura's bridge settings just not count first top layer as bridge, so I can use 100% bridge fan and do not worry about first top layer. So, with *current* implementation super slicer can work like a cura

But I want super slicer to work a little better, to be able set any speed of top layers, including first top layer

**Comment by supermerill** [comment]
Actually, the top fan speed is only activated for the last top layer.
Having control of the internal bridge fan isn't enough.

**Comment by CageFox** [comment]
> 
> 
> Actually, the top fan speed is only activated for the last top layer. Having control of the internal bridge fan isn't enough.

This is additional think to be considered later ;)

For now:

1. I need this feature (disable fan for internal bridge or set another speed, lower then default)
2. I see no drawbacks if it will be implemented

The fan setting for internal bridge originally implemented by my request. Thank you for this.

When I created request I never can imagine this will be implemented with such strange restriction. And I ask you to remove this restriction. Maybe there is so little opinions here because no one using this new feature as it is's usefullness with this restriction is ... restricted

**Comment by MonsterEnergy39** [comment]
I just want to get up to speed with this, as I recently switched to SuperSlicer and my fan is ALWAYS OFF since then.
Where can I find those fan settings?

Using v2.3.57 but can't find it in any of the print/printer-settings

**Comment by supermerill** [comment]
@MonsterEnergy39 
filament settings -> cooling
or clic on a magnifying glass icon and type "fan speed" or whatever you want to find.

**Comment by MonsterEnergy39** [comment]
@supermerill I still have to figure out a lot, but WOW was that obvious :D
Thanks!

**Comment by Woogoo336** [comment]
Found this post looking for an issue I'm having. I was preparing a cooling preset for ASA when I notice it says fan speed will be 90% for internal bridges. I can't find this setting anywhere, and looking at this post it looks like it might be a remnant of an old setting or something strange.

![Image](https://github.com/user-attachments/assets/508cec03-74cd-4b93-9b3a-fe16c830b096)
